sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -336,28 +319,98 @@ function setName(entry: any, nameProperties: any): any {
- *
- * @param errorBody
+ * Utility to help construct the normalized `RestError` object based on given error
+ * information and other data present in the received `response` object.
  * @param response
  */
-export function buildError(errorBody: any, response: HttpOperationResponse): RestError {
-  const errorProperties = errorBody.Error || errorBody.error || errorBody;
-  let errorMessage;
+export function buildError(response: HttpOperationResponse): RestError {
+  if (!isKnownResponseCode(response.status)) {
+    throw new RestError(
+      `Service returned an error response with an unrecognized HTTP status code - ${response.status}`,
+      "ServiceError",
+      response.status,
+      stripRequest(response.request),
+      stripResponse(response)
+    );
+  }
 
+  const errorBody = response.parsedBody;
+  let errorMessage;
   if (typeof errorBody === "string") {
     errorMessage = errorBody;
   } else {
-    errorMessage = errorProperties.Detail || errorProperties.detail;
+    if (
+      errorBody == undefined ||
+      errorBody.Error == undefined ||
+      errorBody.Error.Detail == undefined
+    ) {
+      errorMessage =
+        "Detailed error message information not available. Look at the 'code' property on error for more information.";
+    } else {
+      errorMessage = errorBody.Error.Detail;
+    }
   }
 
+  const errorCode = getErrorCode(response, errorMessage);
+
   const error: RestError = new RestError(
     errorMessage,
-    errorProperties.Code || errorProperties.code,
+    errorCode,
     response.status,
     stripRequest(response.request),
     stripResponse(response)
   );
   return error;
 }
+
+/**
+ * @ignore
+ * Helper utility to construct user friendly error codes based on based on given error
+ * information and other data present in the received `response` object.
+ * @param response
+ * @param errorMessage
+ */
+function getErrorCode(response: HttpOperationResponse, errorMessage: string): string {
+  if (response.status == 401) {
+    return "UnauthorizedRequestError";
+  }
+  if (response.status == 404) {
+    return "MessageEntityNotFoundError";
+  }
+  if (response.status == 409) {
+    if (response.request.method == "DELETE") {
+      return "ServiceError";
+    }
+
+    if (response.request.method == "PUT" && response.request.headers.get("If-Match") == "*") {
+      return "ServiceError";
+    }
+
+    if (errorMessage && errorMessage.toLowerCase().includes("subcode=40901")) {
+      return "ServiceError";
+    }
+
+    return "MessageEntityAlreadyExistsError";
+  }
+
+  if (response.status == 403) {
+    if (errorMessage && errorMessage.toLowerCase().includes("subcode=40301")) {
+      return "InvalidOperationError";
+    }
+    return "QuotaExceededError";
+  }
+
+  if (response.status == 400) {
+    return "ServiceError";
+  }
+
+  if (response.status == 503) {
+    return "ServerBusyError";
+  }
+
+  if (isKnownResponseCode(response.status)) {
```

Comment:
If the status is not a known response code, then we are throwing a RestError right away in `buildError()`. Therefore, we can skip the `isKnownResponseCode` check here.

Reply:
Agree, updated